src/components/panel/panel.js

```diff
@@ -832,6 +838,82 @@ MdPanelRef.prototype._addStyles = function() {
 };
 
 
+/**
+ * Listen for escape keys and outside clicks to auto close.
+ * @private
+ */
+MdPanelRef.prototype._activeListeners = function() {
```

Comment:
How about breaking this into two methods like `configureEscapeToClose` and `configureClickOutsideToClose` ?